assets/apps/dashboard/src/Components/Plugin/InstallActivate.js

```diff
@@ -0,0 +1,182 @@
+/* global neveDash */
+import { get } from '../../utils/rest';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Button, Dashicon } from '@wordpress/components';
+import { useEffect } from 'react';
+
+const InstallActivate = ({
+	name = '',
+	labels = [],
+	slug,
+	successActivation,
+	autoInstall = false,
+	smallButton = false,
+	pluginState,
+	activateURL,
+	description,
+	// pluginBasename,
+	// successUpdate,
+}) => {
+	const { getPluginStateBaseURL, pluginsURL } = neveDash;
+
+	const [installing, setInstalling] = useState(false);
+	const [activating, setActivating] = useState(false);
```

Comment:
Not necessary but I feel it would simplify things a bit - we can probably abstract these two states into a single one called progress.
```suggestion
	const [progress, setProgress] = useState(false);
```

Then we can set it subsequently in the actions: `false > installing > activating`.

Again, it's just a suggestion, not necessary.